drivers/mipi_dbi/mipi_dbi_stm32_fmc.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2024 Bootlin
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_fmc_mipi_dbi
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/mipi_dbi.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(mipi_dbi_stm32_fmc, CONFIG_MIPI_DBI_LOG_LEVEL);
+
+#define FMC_DATA_ADDR_OFFSET 0x02
+
+struct mipi_dbi_stm32_fmc_config {
+	/* Reset GPIO */
+	const struct gpio_dt_spec reset;
+	/* Power GPIO */
+	const struct gpio_dt_spec power;
+	uint16_t *register_addr;
+	uint16_t *data_addr;
+};
+
+struct mipi_dbi_stm32_fmc_data {
+};
+
+int mipi_dbi_stm32_fmc_command_write(const struct device *dev,
+				     const struct mipi_dbi_config *dbi_config, uint8_t cmd,
+				     const uint8_t *data_buf, size_t len)
+{
+	const struct mipi_dbi_stm32_fmc_config *config = dev->config;
+
+	uint32_t i = 0;
+
+	*config->register_addr = cmd;
+
+	while (i < len) {
+		uint16_t data = (uint16_t)data_buf[i];
+		*config->data_addr = data;
+		i += 1;
+	}
+
+	return 0;
+}
+
+static int mipi_dbi_stm32_fmc_write_display(const struct device *dev,
+					    const struct mipi_dbi_config *dbi_config,
+					    const uint8_t *framebuf,
+					    struct display_buffer_descriptor *desc,
+					    enum display_pixel_format pixfmt)
+{
+	const struct mipi_dbi_stm32_fmc_config *config = dev->config;
+	uint32_t i = 0;
+
+	while (i < desc->buf_size) {
+		uint16_t data = ((uint16_t)framebuf[i + 1U] << 8U) | (uint16_t)framebuf[i];
+		*config->data_addr = data;
+		i += 2;
+	}
+
+	return 0;
+}
+
+static int mipi_dbi_stm32_fmc_reset(const struct device *dev, uint32_t delay)
+{
+	const struct mipi_dbi_stm32_fmc_config *config = dev->config;
+	int ret;
+
+	if (config->reset.port) {
```

Comment:
```suggestion
	if (config->reset.port == NULL) {
```

Reply:
Thanks for the catch ! I think I wanted to write `!config->reset.port` but forgot the `!`